src/simpleweighteddigraph.jl

```diff
@@ -105,13 +117,39 @@ function add_edge!(g::SimpleWeightedDiGraph, e::SimpleWeightedGraphEdge)
     return true
 end
 
-function rem_edge!(g::SimpleWeightedDiGraph, e::SimpleWeightedGraphEdge)
-    has_edge(g, e) || return false
-    U = weighttype(g)
-    @inbounds g.weights[dst(e), src(e)] = zero(U)
+rem_edge!(g::SimpleWeightedDiGraph{T, U}, e::AbstractEdge) where {T<:Integer, U<:Real} =
```

Comment:
Can't this be an interface method?

Reply:
I'm not sure we want to enforce `AbstractSimpleWeightedGraph` to be mutable, so I left it here. I we add a type or a trait for mutable graph, we should envision it.